package.json

```diff
@@ -869,15 +869,16 @@
 		}
 	  },
 	"scripts": {
-		"postinstall": "cd server && npm install && cd ../client && npm install && cd ..",
+		"build:wasm": "bash ./scripts/build-tree-sitter-wasm.sh",
+		"postinstall": "npm run build:wasm && cd server && npm install && cd ../client && npm install && cd ..",
```

Comment:
`build:wasm` doesn't have to be in the `postinstall`. There is a case where you only run `npm install` to make sure the packages are installed and the wasm files don't need to be rebuilt.

Reply:
You're right it does not have to be in postinstall and there are valid cases we would not want it there. However, I think it is also valid to expect `npm install` to install all the dependencies of the project. Keeping one single step makes it easier for new comers and reduce the chances they'll miss something and waste time on trying to make it work.

Maybe we could add a hash to the .info files and avoid regenerating the wasm file if it corresponds to that hash, but I am not sure how reliable it would be and if it is worth the effort.

What do you think?

Reply:
My opinion is if there is a valid reason not to put it inside `postinstall` then don't put it. Since you are documenting this new process, you could mention it in the [README for devs](README-DEVELOPER.md). I think it is also nice to reflect this in the CI because it is now a critical step to build the extension.

Alternatively, you add a check in the script to see if the wasm files exist and build wasms if they don't. If they do, then compare the references in the .info file with the remote to if a rebuild is required.

Reply:
I see. yarn explicitly [recommends not using postinstall](https://yarnpkg.com/advanced/lifecycle-scripts#postinstall) since it could be used to execute malicious scripts. Lets remove.

Reply:
Done.